src/Avalonia.Controls/Navigation/INavigationRouter.cs

```diff
@@ -0,0 +1,49 @@
+﻿using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Avalonia.Controls
+{
+    public interface INavigationRouter : INotifyPropertyChanged
```

Comment:
Why force to implement INotifyPropertyChanged when deriving INavigationRouter? There may be some contexts where it is not needed.

Reply:
This is used by the control to update the ContentView property when the router changes it. An alternative would be to use an action or event, but that would rely more on the user calling those events. Requiring to implement `INotifyPropertyChanged` will ensure the control gets notified.